k4FWCore/src/PodioDataSvc.cpp

```diff
@@ -109,10 +126,16 @@ StatusCode PodioDataSvc::i_setRoot(std::string root_path, DataObject* pRootObj)
 }
 
 void PodioDataSvc::endOfRead() {
+  m_eventNum++;
+
+  if (!m_bounds_check_needed) {
+    return;
+  }
```

Comment:
```suggestion

  if (!m_bounds_check_needed) {
    return;
  }
  
    m_eventNum++;
```
if it's going to return we don't need to increment anything

Reply:
We always need to increment the event number (at least if we aren't in the last event yet). This change would cause us to read the same event again and again.

Reply:
Ah indeed, I thought https://github.com/AIDASoft/podio/blob/master/include/podio/ROOTFrameReader.h#L83 was being used, I think that would be better than having to worry about counting since it's always going to be the next one

Reply:
Even then we would still need to do the current counting as we need to issue the stop to gaudi before we actually hit the non-existing event as far as I understand